frontend/src/components/core/MainMenu/MainMenu.tsx

```diff
@@ -517,6 +525,23 @@ function MainMenu(props: Props): ReactElement {
 
   const { hostIsOwner } = props
 
+  let showDevelopmentMenu: boolean
+  if (props.toolbarMode == Config.ToolbarMode.DEVELOPER) {
+    showDevelopmentMenu = true
+  } else if (
+    props.toolbarMode == Config.ToolbarMode.VIEWER ||
+    props.toolbarMode == Config.ToolbarMode.MINIMAL
+  ) {
+    showDevelopmentMenu = false
+  } else {
+    showDevelopmentMenu = hostIsOwner || isLocalhost()
+  }
+
+  if (menuItems.length == 0 && !showDevelopmentMenu) {
+    // Don't show an empty menu.
+    return <></>
+  }
```

Comment:
every time you use `let` a panda is crying.

```
function showDeveloperMenu(){
    if (props.toolbarMode == Config.ToolbarMode.DEVELOPER) {
        return true
    }
    if (
    props.toolbarMode == Config.ToolbarMode.VIEWER ||
    props.toolbarMode == Config.ToolbarMode.MINIMAL
  ) {
         return false;
     }

     return hostIsOwner || isLocalhost()
}
```

Reply:
Thanks for suggestion.
I added two commits to my branch.  One creates functions to avoid creating `let` variable, and the second refactors the code a bit more so that we don't need to calculate the menu items when we know the menu is hidden.

Reply:
Thanks for suggestion.
I added two commits to my branch.  One creates functions to avoid creating `let` variable, and the second refactors the code a bit more so that we don't need to calculate the menu items when we know the menu is hidden.